pkgs/tools/security/wordlists/default.nix

```diff
@@ -0,0 +1,68 @@
+{ lib
+, pkgs
+, runtimeShell
+, symlinkJoin
+, tree
+}:
+
+let
+  _wordlistData =
+    [ { name = "dirbuster"; expressionPath = ./dirbuster.nix; }
+      { name = "nmap"; expressionPath = ./nmap.nix; needsPackage = true; }
+      { name = "rockyou"; expressionPath = ./rockyou.nix; }
+      { name = "seclists"; expressionPath = ./seclists.nix; }
+      { name = "wfuzz"; expressionPath = ./wfuzz.nix; needsPackage = true; }
+    ];
+
+  scopedWordlists = lib.makeScope pkgs.newScope (self: with self;
+    let _packageWithLinkCommandFrom = acc: datum:
+      # Some wordlists are based on an already existing `nixpkgs` package; Pass it as parameter of the expression.
+      let parameters = {}
+        // lib.optionalAttrs ((builtins.elem "needsPackage" (builtins.attrNames datum)) && datum.needsPackage) { ${datum.name} = pkgs.${datum.name}; };
+      in
+        let p = callPackage datum.expressionPath parameters;
+        in acc // { ${datum.name} = p; };
+    in builtins.foldl' _packageWithLinkCommandFrom {} _wordlistData
+  );
+
+in let
+  packageConstructor = wordlistPackages:
+    symlinkJoin rec {
+      pname = "security-wordlists";
+      version = "unstable-2020-11-23";
+
+      name = "${pname}-${version}";
+
+      paths = wordlistPackages;
+
+      dontUnpack = true;
+
+      propagatedBuildInputs = [ tree ];
```

Comment:
why is this necessary?

Reply:
Because `tree` is a "runtime" dependency of this package; See 7 lines below.

Reply:
My reasoning for this:
 * `tree` is referenced directly in the builder, so we don't need to specify it at build-time.
 * `tree` is present in cleartext in `$out`, so nix will find that reference and correctly deal with it.
Thus we don't need to specify it in `propagatedBuildInputs`